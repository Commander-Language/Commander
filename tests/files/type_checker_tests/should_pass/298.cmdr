//Test builtin parseBool type checks
a = 1;
x = parseBool(1);
x = parseBool(3.0);
x = parseBool(true);
x = parseBool("true");
x = parseBool((1.0, true, 3)[a]);
