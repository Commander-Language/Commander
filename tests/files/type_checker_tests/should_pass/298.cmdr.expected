(PrgmNode
	(ExprStmtNode (BinOpExprNode (BindingNode a) SET (IntExprNode 1 INT) INT))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode parseBool (fn (INT) -> BOOL)) (fn (INT) -> BOOL)) (ExprsNode (IntExprNode 1 INT)) BOOL) BOOL))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode parseBool (fn (FLOAT) -> BOOL)) (fn (FLOAT) -> BOOL)) (ExprsNode (FloatExprNode 3.000000 FLOAT)) BOOL) BOOL))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode parseBool (fn (BOOL) -> BOOL)) (fn (BOOL) -> BOOL)) (ExprsNode (BoolExprNode true BOOL)) BOOL) BOOL))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode parseBool (fn (STRING) -> BOOL)) (fn (STRING) -> BOOL)) (ExprsNode (StringExpressionNode (StringNode (StringExprsNode (StringExpressionNode (StringNode 'true') STRING))) STRING)) BOOL) BOOL))
	(ExprStmtNode (BinOpExprNode (BindingNode x) SET (CallExprNode (LValueExprNode (VarExprNode parseBool (fn (INT) -> BOOL)) (fn (INT) -> BOOL)) (ExprsNode (IndexExprNode (TupleExprNode (ExprsNode (FloatExprNode 1.000000 FLOAT) (BoolExprNode true BOOL) (IntExprNode 3 INT)) (FLOAT, BOOL, INT)) (LValueExprNode (VarExprNode a INT) INT))) BOOL) BOOL)))
